File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> complement_map;

        for(int j = 0; j < nums.size(); j++){
            if(complement_map.count(target - nums[j])){
                return {complement_map[target - nums[j]], j};
            }
            complement_map[nums[j]] = j;
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if(len(numbers) == 2):
            return [1,2]

        l = 0
        r = len(numbers) - 1
        while(l < r):
            add = numbers[l] + numbers[r]
            res = add - target

            if(res == 0):
                return [l+1, r+1]
            if(res > 0):
                r-=1
            else:
                l+=1

        return []
